### PR TITLE
Added completed handler - TODO add actual completed animation

### PR DESCRIPTION
**Pull request by geea-develop**
Screenshot 2024-07-18 at 23 21 37

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -10,26 +10,32 @@ export type SquareClasses =
 export type SquareSize = "small" | "big";
 
 export type SelectedSquare = {
+    id: string;
+    image: string;
+    setClassName?: SetStateFunction<SquareClasses>;
+};
+
+export type NullableSelectedSquare = {
```
Comment (YoadBarok) [inline src/types.ts]: Maybe we should just create a generic `Nullable` type that can be reused like:
`export type Nullable<T> = T | null;` 
That way we can use it as: `Nullable<SelectedSquare>`

File: src/components/board.tsx [hunk 2]
```diff
@@ -13,19 +13,42 @@ type Props = {
...
 
   const [getSecondSelectedSquare, setSecondSelectedSquare] =
-    useState<SelectedSquare | null>(null);
+      useState<NullableSelectedSquare>(null);
 
+  useEffect(() => {
+      const values = Array.from(squaresMap.values());
+      const isCompleted = values
+          .map((k) => k.getClassName === "matched")
+          .every((k) => k);
```
Comment (YoadBarok) [inline src/components/board.tsx]: Why not directly:
`values.every((square) => square.getClassName === "matched")`?

File: src/components/board.tsx [hunk 3]
```diff
@@ -13,19 +13,42 @@ type Props = {
...
+        squaresMap.forEach((square) => {
+            counter++;
+            setTimeout(() => {
+                square.setContent?.(`👍`);
+            }, counter * 100);
+        });
+      }
+  }, [change, squaresMap]);
+
+  const handleChange = useCallback(() => {
```
Comment (YoadBarok) [inline src/components/board.tsx:24]: Nice one!

File: src/hooks/matchChecker.ts [hunk 4]
```diff
@@ -1,51 +1,65 @@
...
+    secondSelectedSquare: NullableSelectedSquare;
+    squaresMap: SquareMap;
+    setClickEnabled: SetStateFunction<boolean>;
+    setFirstSelectedSquare: SetStateFunction<NullableSelectedSquare>;
+    setSecondSelectedSquare: SetStateFunction<NullableSelectedSquare>;
+    handleChange: CallableFunction;
+}) => {
+    useEffect(() => {
+        const checkForMatch = (
+            firstSquare: SelectedSquare,
```
Comment (YoadBarok) [inline src/hooks/matchChecker.ts:23]: Nice idea adding these params to avoid the null check that we had before!

**Review by YoadBarok — CHANGES_REQUESTED** [review]

File: src/components/board.tsx [hunk 5]
```diff
@@ -13,19 +13,42 @@ type Props = {
...
+  const [change, setChange] = useState(0);
   const [getFirstSelectedSquare, setFirstSelectedSquare] =
-    useState<SelectedSquare | null>(null);
+    useState<NullableSelectedSquare>(null);
 
   const [getSecondSelectedSquare, setSecondSelectedSquare] =
-    useState<SelectedSquare | null>(null);
+      useState<NullableSelectedSquare>(null);
 
+  useEffect(() => {
```
Comment (YoadBarok) [inline src/components/board.tsx]: Maybe we can separate this logic to another hook file similar to how matchChecker.ts is?

File: src/types.ts (hunk 1 above)
Comment (geea-develop) [reply src/types.ts]: done

File: src/components/board.tsx (hunk 2 above)
Comment (geea-develop) [reply src/components/board.tsx]: done

File: src/components/board.tsx (hunk 5 above)
Comment (geea-develop) [reply src/components/board.tsx]: done

**Review by YoadBarok — APPROVED** [review]

**Pull request merged**